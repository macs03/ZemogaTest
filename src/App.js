import React, {useEffect, useState} from 'react';
import {StatusBar, View, StyleSheet} from 'react-native';
import SplashScreen from 'react-native-splash-screen';

import app from './lib/app';
import MainNavigator from './navigation/mainNavigator';
import Splash from './screens/splash/splash';

const App = () => {
  const [isBooting, setIsBooting] = useState(true);

  useEffect(() => {
    app.boot().then(() => {
      SplashScreen.hide();
      setIsBooting(false);
    });
  }, []);

  if (isBooting) {
    return <Splash />;
  }

  return (
    <>
      <View style={styles.content}>
        <StatusBar barStyle="light-content" />
        <MainNavigator />
      </View>
    </>
  );
};

const styles = StyleSheet.create({
  content: {height: '100%'},
});

export default App;
