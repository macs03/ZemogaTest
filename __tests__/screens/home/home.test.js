/**
 * @format
 */

import 'react-native';
import React from 'react';
import Home from '../../../src/screens/home/home';
import App from '../../../src/lib/app';

import renderer from 'react-test-renderer';

beforeEach(() => {
  App.bootMocked();
});

it('renders correctly', () => {
  App.api.get = jest.fn().mockImplementation(() => Promise.reject('fail'));

  const tree = renderer.create(
    <Home
      navigation={{
        setOptions: () => {},
      }}
      route={{}}
    />,
  );
  expect(tree).toMatchSnapshot();
});

it('renders correctly with data from service', async () => {
  App.api.get = jest.fn().mockImplementation(() =>
    Promise.resolve({
      id: 1,
    }),
  );

  const tree = renderer.create(
    <Home
      navigation={{
        setOptions: () => {},
      }}
      route={{}}
    />,
  );
  expect(tree).toMatchSnapshot();
});
