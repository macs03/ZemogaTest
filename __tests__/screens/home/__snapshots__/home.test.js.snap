// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "height": "100%",
      "justifyContent": "flex-start",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FAFAFA",
        "flexDirection": "row",
        "height": 50,
        "justifyContent": "space-around",
        "paddingHorizontal": 20,
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#16AA16",
          "borderColor": "#16AA16",
          "borderRadius": 2,
          "borderWidth": 1,
          "height": "60%",
          "justifyContent": "center",
          "opacity": 1,
          "width": "50%",
        }
      }
      testID="filtering-0"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#16AA16",
            },
            Object {
              "color": "#FFFFFF",
            },
          ]
        }
      >
        All
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#16AA16",
          "borderRadius": 2,
          "borderWidth": 1,
          "height": "60%",
          "justifyContent": "center",
          "opacity": 1,
          "width": "50%",
        }
      }
      testID="filtering-1"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#16AA16",
            },
            false,
          ]
        }
      >
        Favorite
      </Text>
    </View>
  </View>
  <Text>
    There is nothing
  </Text>
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#D6000C",
          "height": 80,
          "justifyContent": "center",
          "opacity": 1,
          "width": "100%",
        }
      }
      testID="deleteButton"
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
          }
        }
      >
        Delete All
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders correctly with data from service 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "height": "100%",
      "justifyContent": "flex-start",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FAFAFA",
        "flexDirection": "row",
        "height": 50,
        "justifyContent": "space-around",
        "paddingHorizontal": 20,
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#16AA16",
          "borderColor": "#16AA16",
          "borderRadius": 2,
          "borderWidth": 1,
          "height": "60%",
          "justifyContent": "center",
          "opacity": 1,
          "width": "50%",
        }
      }
      testID="filtering-0"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#16AA16",
            },
            Object {
              "color": "#FFFFFF",
            },
          ]
        }
      >
        All
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#16AA16",
          "borderRadius": 2,
          "borderWidth": 1,
          "height": "60%",
          "justifyContent": "center",
          "opacity": 1,
          "width": "50%",
        }
      }
      testID="filtering-1"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#16AA16",
            },
            false,
          ]
        }
      >
        Favorite
      </Text>
    </View>
  </View>
  <Text>
    There is nothing
  </Text>
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#D6000C",
          "height": 80,
          "justifyContent": "center",
          "opacity": 1,
          "width": "100%",
        }
      }
      testID="deleteButton"
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
          }
        }
      >
        Delete All
      </Text>
    </View>
  </View>
</View>
`;
